feat: build from tacc docs docker image

wesleyboar opened the pull request:
## Overview

Give https://designsafe-ci.org/user-guide/ the theme and features of https://docs.tacc.utexas.edu/ in such a way that other repos can do the same with ease.

### Status

Reviewed multiple times by developer, designer, stakeholder, and contributor. Stakeholder says to deploy when I feel comfortable. I will merge into `epic/redesign-2024`, so further changes are independent of this TACC-Docs PR.

<details>

- [x] Developer review \# 1.
- [x] Developer review \# 1 updates.
    <sup>#2</sup>
- [x] Stakeholder review \# 1.
- [x] Stakeholder review \# 1 updates.
    <sup>57f498f</sup>
- [x] Deploy.
- [x] Document intent to make installable TACC theme.
- [x] Use a TACC-Docs version **not** a commit (pre-release of [TACC/TACC-Docs#15](https://github.com/TACC/TACC-Docs/pull/15)).
- [x] [Developer review \# 2](https://github.com/DesignSafe-CI/DS-User-Guide/pull/1#pullrequestreview-2041777918).
- [x] Developer review \# 2 updates.
    <sup>https://github.com/DesignSafe-CI/DS-User-Guide/pull/1#pullrequestreview-2094707095
    https://github.com/DesignSafe-CI/DS-User-Guide/pull/1#pullrequestreview-2069087570</sup>
- [x] Stakeholder review \# 2. — 2024/05/28
    <sup>86c4878, get "Contributor review \#1"</sup>
- [x] Contributor review \# 1 (@silviamazzoni).
- [x] Contributor review \# 1 updates.
   <sup>51d1367, 6bca8d4, editor can instead use [old nav config](https://github.com/DesignSafe-CI/DS-User-Guide/blob/816dcd1/user-guide/mkdocs.yml#L62-L85) to set title for navigation</sup>
- ~~Stakeholder review \# 3.~~
    <sup>I was told I could deploy when I think it is ready.</sup>
- [x] Swap out the logo with [new one](https://www.designsafe-ci.org/).
    <sup>3c562755, e29e2bc6</sup>

</details>

## Related

- [DES-2469](https://tacc-main.atlassian.net/browse/DES-2469)
- requires https://github.com/TACC/TACC-Docs/pull/15
- depends on #14

## Changes

- **added** rule to ignore [TACC-Docs](https://github.com/TACC/TACC-Docs) files in Git
- **changed** Dockerfile to only:
    - load taccwma/tacc-docs
    - archive [TACC-Docs](https://github.com/TACC/TACC-Docs) files
    - copy over specific [TACC-Docs](https://github.com/TACC/TACC-Docs) files
- **changed** Makefile to remove TACC assets only for Docker users
- **documented** local development gotcha
- **added** script to clone essential [TACC-Docs](https://github.com/TACC/TACC-Docs) files
- **added** code to integrate TACC docs only for Docker users
- **removed** Poetry dependencies (because TACC will provide them)
- **removed** custom styles and scripts (because TACC will provide them)
- **added** favicon, logo, and CMS link

## Testing

### Live

> [!TIP]
> Available on DesignSafe "Next" server at `/user-guide/`.

### TACC Theme via Docker

1. [Setup site via Docker](https://github.com/DesignSafe-CI/DS-User-Guide/blob/task/tacc-docs-shared-docker/README.md#b-via-docker) **or** see deploy works at https://pprd.docs.tacc.utexas.edu/.
2. Verify site has style of https://docs.tacc.utexas.edu/.

### Non-Docker Regression Test

1. [Setup site via Python](https://github.com/DesignSafe-CI/DS-User-Guide/blob/task/tacc-docs-shared-docker/README.md#a-via-python)
2. Verify site has [MkDocs readthedocs theme](https://www.mkdocs.org/user-guide/choosing-your-theme/#readthedocs).

### Docker & Non-Docker Steps

1. Verify site has content of https://designsafe-ci.org/user-guide/.
2. Verify pages load images.
3. Verify search still works.
4. Verify Nav behavior matches production.

## UI

| TACC Theme via Docker— as of 51d1367 |
| - |
| <img width="1200" alt="via docker" src="https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/a7424435-0468-4235-b5ff-21b06d075ffe"> |

| Non-Docker Regression Test — as of 412eff9 |
| - |
| <img width="1200" alt="via python" src="https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/7c14a098-92af-4c1b-b935-269b3ba8e55a"> |

## Notes

Installing the TACC assets and theme dir as a MkDocs theme is a **great** idea! See [CMD-154: TACC Docs Styles via Installable Theme](https://tacc-main.atlassian.net/browse/CMD-154). Not done yet, because TACC Docs theme might because there is a deadline to meet.

[huetone.ardov.me]: https://huetone.ardov.me/?palette=N4IgdghgtgpiBcIAqBBAwmkAaEALArjAM4IDaoksCIEA9AIy0AOA9gE4AuEANtiAMYtu7EvFIgAxADMIUgCxSpfafRgAjKQBNlMAMyaA7FuWaArBsXKIugBwGY2nBIM2FMG8oBsABl0BOfn5lfXobPzVlACYbem9PT2VVb345IIBdAF8sCmg4RDpGfigSHEFhNlFxaV0pIzgnTSkYWXrJfghNRtaJCAh%2BIyUnG08IGEsnb19-P2VJyMj42e96el1dWcj6SN1vJe8IeZBM7PBc6jpIvjKRMkkbb3v75UeH3acpfilPGCCnGFNDPw3pIzGpTAZfncDAY5KZ1k5TPQ5LposEUfx6IM7q8nkMcbtjjkqPlaPCBEIbmJsS9nvidEZMcCJGpOg4Ik4-JFAvxTF56NYAso5HFNJ4PE5VGtdPRaTS8TTCadiSAAOZsCAATyuFIqtwkL1x1NesuN7wU5pMBkMhmUBm8drtQtczqi8zdJsN%2BrpisoeRAHFobAc2vKlSNnoNTMjyg%2BMDkmnZkg%2Bsj8Mr%2BmlSK2U-G2EF8yj8KLUcQ9Ue9WSJfoDGpg3GEAHcQ5SqtH5abw0ytFJvJoydIHDZPVJImoUQZs5o-AZ6JCJH4-J5uxBSyufWdEAG1TAYGAm7qqV65R2V041LIbJpZ4WvvxHJJYrUIDMnPp%2BDBOVExTZTFjD%2B2-08a7KgGajcIQe5hgBZZHlBMbeF84xtJECYyMEUjfvIUSmOYahpveMJPs%2Bx5toBFZKn6Fy0JoJTkqGeqtsRbSNH4mFOBAphxp0zypD8CQIjYED0OEwQuPQBiJhIyHmrODGwUBFEMFRNHXPuVTblILGzp8Py1MoYJ6Bpzy6G%2BqEIiMb7ji%2BugGBAaiWZI0Q%2BH4TKRN48Ywok8xqMKswYmokQzPJ5ykkpEH0XSJFMj8cYaPSmhhPZEgZhii5WGE7ESfcBieHIRESNhCgoiex5BRutBbjuoWlDqkGyXVEWSB0CbWLachgosThyLlGZyMEfiuP4WGmJ4kS8pFq5kb61DVrWDZVbRzaMbB41-G%2BbW-jAVoJdm-L-DJ9o5XlBVSEVK0lRkaQ4BwLBgMQtxQBAAAeAC03AAJYqrgHB8I9v2vR9X0-S972fd9ODAwDYMgCDgM4GA7APbwOCaBAbAANY-c9KPo0DWOoxj4N-djGOZEAA

wesleyboar committed:
feat: only put in Docker container what it needs

I.e. Contianer now does not have .git nor .github nor Dockerfile, etc.

wesleyboar committed:
feat: use TACC/TACC-Docs Docker image

wesleyboar committed:
fix: Makefile does not need docker commands

wesleyboar committed:
feat: build off of taccwma matching docker image

wesleyboar committed:
docs: add commands to dev on tacc styles version

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
feat(docker): sync docs yet retain tacc code

1. Sync docs via volumes.
2. Restore TACC code that is wiped out.
3. Do not sync TACC code to host.

wesleyboar committed:
feat(docker): sync mkdocs.yml

wesleyboar committed:
fix(docker): do not remove /code-from-tacc

The copy commands fail after container is stopped.

Removing it is nice for storage space, but perhaps negligible.

Keeping it will avoid confusion for anyone new to debugging containers.

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
feat: load favicon and logo from designsafe-ci.org

wesleyboar committed:
test: sync mkdocs config so as to merge tacc's

wesleyboar committed:
chore(docker): tweak comments

wesleyboar committed:
docs: mkdocs site_favicon

wesleyboar committed:
fix: integrate TACC code

wesleyboar committed:
fix: cms name and link

wesleyboar committed:
chore: do not ignore what does not (yet) exist

wesleyboar committed:
docs: uppercase a "not" to emphasize it

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
feat: poetry 1.2.0 → 1.4.2 (#2)

wesleyboar committed:
feat: minimal Makefile (#3)

wesleyboar committed:
feat: nav bar via mega docs using snippets / includes (#4)

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'dev' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'dev' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'dev' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
Merge branch 'main' into dev

wesleyboar committed:
⚠️ Merge branch 'main' into task/tacc-docs-shared-docker

I retained the poetry.lock and pyproject.toml that were deleted in the branch and added into main.

A plugin that TACC Docs still uses was replaced by a new one in DS Docs that TACC DOcs doesn't have…

I dunno what to do yet. At the least, I document in this commit, what's going on.

wesleyboar committed:
Revert "⚠️ Merge branch 'main' into task/tacc-docs-shared-docker"

This reverts commit 9ba0d5151058432dc9da2d362147447fad1a1315, reversing
changes made to 2d35e33d679a3cd1ffbe5646f4a07fc0b652a8f2.

wesleyboar committed:
Merge branch 'dev' into task/tacc-docs-shared-docker

wesleyboar commented:
Accidentally closed. Reopening. I am scheduled to work on this again this month.

rstijerina commented on user-guide/docs/analysis/overview.md:
```diff
@@ -1,5 +1,14 @@
+### [Status Updates](#status) { #status }
+
+<ul>
+	<li>HVSR web -  Last updated 04/20</li>
```
Is it good practice here to hard code last updated dates?

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar commented:
Migrated to #18 (because GitHub closed this PR because I prematurely deleted `dev` branch).

wesleyboar committed:
fix: restores file edits presumedly from bad merge

wesleyboar committed:
fix: recover css file edit that was actually good

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
feat: mkdocs.yml.sh → INHERIT config

Mirror: https://github.com/TACC/TACC-Docs/pull/15/commits/788ac8ec

wesleyboar committed:
feat: mkdocs.yml.sh → INHERIT config

Mirror: https://github.com/TACC/TACC-Docs/pull/15/commits/788ac8ec

wesleyboar committed:
fix: seismic page no content

wesleyboar committed:
fix: inc. paths ie use include-markdown not macros

wesleyboar committed:
chore: revert an unnecessary image path change

wesleyboar committed:
fix: revert now unnecessary image path changes

wesleyboar committed:
chore: revert unnecessary newline deletions

wesleyboar committed:
fix: use of local docker image by using remote one

wesleyboar committed:
test: duplicate TACC-Docs mkdocs.base.yml

wesleyboar committed:
test: duplicate TACC-Docs poetry and pyproject

wesleyboar committed:
feat: support Local Development via Python

wesleyboar committed:
docs: major update requires dependency install

wesleyboar committed:
chore: delete unused complexity

wesleyboar committed:
docs: warn users local dev has diff theme

wesleyboar committed:
quick: restore accidentally deleted file

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
refactor: des-2469 simplify (DRAFT)

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'task/tacc-docs-shared-docker' into task/tacc-docs-shared-docker--wes-n-sal

wesleyboar committed:
refactor: simpler way to ensure a tacc-readthedocs

wesleyboar committed:
fix: ignore TACC files

wesleyboar committed:
chore: ignore TACC files and dirs

wesleyboar committed:
fix: build with tacc assets & do not lose them

wesleyboar committed:
style: cleaner diff

wesleyboar committed:
Merge pull request #2 from DesignSafe-CI/task/tacc-docs-shared-docker--wes-n-sal

feat: build from tacc docs docker image - refactor: des-2469 simplify

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
refactor: move autoScrollNav.js to core

wesleyboar committed:
feat: use latest TACC docs image for this

wesleyboar committed:
feat: use latest TACC docs image for this

wesleyboar committed:
chore: remove markdown_extensions cuz TACC has 'em

wesleyboar committed:
chore: remove extra_javascript cuz TACC has 'em

wesleyboar committed:
fix: prevent docker dir copy err by adding files

These files force the directories to exist.

wesleyboar committed:
feat: use nav_redirects from TACC-Docs

wesleyboar committed:
fix: should not need tacc-setup.sh for docker

wesleyboar committed:
fix: latest TACC-Docs image needs 8-digit tag

wesleyboar committed:
feat: provide TACC/TACC-Docs files on public repo

TACC/TACC-Docs is not public yet.

Lazy option is host on DS in a branch.

Other option is a temporary public repo on wesleyboar or TACC.

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar commented on user-guide/mkdocs.yml:
```diff
@@ -1,39 +1,33 @@
-  - admonition
-  - attr_list
-  - abbr
-  - toc:
-      permalink: True
```
The `toc` plugin is enabled by MkDocs by default ([source a](https://github.com/TACC/TACC-Docs/blob/6b1dd1b/mkdocs.yml#L118-L121), [source b](https://github.com/mkdocs/mkdocs/blob/1.4.2/docs/user-guide/configuration.md#markdown_extensions)).

wesleyboar replied:
K. Cool. But is `permalink: True` the default config?

wesleyboar replied:
No, the default is `False` ([ref.](https://python-markdown.github.io/extensions/toc/#usage)).

But irrelevant, because DS-User-Guide does **not** use [`[TOC]`](https://python-markdown.github.io/extensions/toc/#syntax) ([ref.](https://github.com/search?q=repo%3ADesignSafe-CI%2FDS-User-Guide+%22%5BTOC%5D%22&type=code)).

I tested on [TACC/TACC-Docs which has one instance](https://github.com/TACC/TACC-Docs/blob/2178fe4/docs/code-examples/index.md?plain=1#L3).

| `True` | `False` | default |
| - | - | - |
| <img width="900" alt="True" src="https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/312184c3-ef5d-45e3-ab53-806f69c6a1d4"> | <img width="900" alt="False" src="https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/b8be1db6-ab31-4ad4-b584-89060f528313"> | <img width="900" alt="default" src="https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/a5cf674d-33f5-48ef-b012-0563020d0ad6"> |

wesleyboar commented on user-guide/themes/ds-readthedocs/main.html:
Moved to TACC/TACC-Docs as `nav_redirects`.

wesleyboar commented on user-guide/docs/js/changeNavMarkup.js:
Already in TACC/TACC-Docs.[^1]

[^1]: The `DS_NAVIGATION_REDIRECTS` was recently added to TACC/TACC-Docs as `NAV_REDIRECTS`.

wesleyboar commented on user-guide/docs/js/setTargetForExternalLinks.js:
Already in TACC/TACC-Docs.

wesleyboar commented on user-guide/docs/css/ds-docs.css:
All of these styles are not necessary because they are my short-term fixes done better in TACC/TACC-Docs.

wesleyboar commented on bin/tacc-setup.sh:
How should I make more obvious that this is only for non-Docker development?

wesleyboar commented on README.md:
```diff
@@ -18,12 +18,16 @@ DesignSafe ReadTheDocs Documentation with [MkDocs](https://mkdocs.readthedocs.io
 1. Navigate into your clone of this repo.
 2. Install dependencies:\
-    <sup>You should only need to do this once.</sup>
+    <sup>You should only need to do this once, or after a major update.</sup>
```
```suggestion
    <sup>You should only need to do this once, or after a feature update.</sup>
```

wesleyboar replied:
Resolved: 5ac5e48a

wesleyboar commented on Dockerfile:
```diff
@@ -1,47 +1,15 @@
-
-COPY ./user-guide /docs
-WORKDIR /docs
+FROM taccwma/tacc-docs:e7dfaac4
```
Should be a version… How to version if primary contributor is still making misc. commits to main?

Ah! Maybe versions **only** consider PR changes. Assume every misc. commit to `main` is a "patch".

wesleyboar replied:
Wes says. Wes does. DS-User-Guide uses [TACC-Docs v0.3.1](https://github.com/TACC/TACC-Docs/releases/tag/v0.3.1) since dbadb6f.

wesleyboar committed:
chore: try to reduce diff on EOL

wesleyboar reviewed (commented):
Notes for Reviewer

> **Update**: Most comments "Outdated" by GitHub after 1 commit irrelevant to file on which I commented 😒.

wesleyboar committed:
docs(README): more accurate notice wording

wesleyboar committed:
fix: give logo more horizontal space

wesleyboar committed:
Merge branch 'task/tacc-docs-shared-docker' of github.com:wesleyboar/DS-User-Guide into task/tacc-docs-shared-docker

wesleyboar committed:
fix: new tacc assets not being used

wesleyboar committed:
chore: do not copy unnecessary files from tacc

wesleyboar committed:
feat: add "User Guide" under logo

wesleyboar committed:
feat: DS logo transparent bkgd

wesleyboar commented on Makefile:
```diff
@@ -9,3 +9,6 @@ start:
 .PHONY: stop
 stop:
 	docker-compose -f docker-compose.yml down
+	# So assets are not re-used (even after `docker system prune --all --force`)
+	docker volume rm ds-user-guide_skip-tacc-js
+	docker volume rm ds-user-guide_skip-tacc-css
```
Use a parameter to declare the context (here it is `ds-user-guide`, but only because I cloned the repo to that directory on my machine). You should be able to set the variable `COMPOSE_PROJECT_NAME` to control this.

Reference
- [Core-Portal-Deployments does this.](https://github.com/TACC/Core-Portal-Deployments/blob/7bec278/3dem/camino/pprd.env#L14)
- https://docs.docker.com/compose/environment-variables/envvars/#compose_project_name
- `DOCKER_COMPOSE :=  docker compose ${BASE_COMPOSE} ${COMPOSE_OVERRIDE} --env-file=$(ENV_FILE)`

wesleyboar replied:
Fixed via e8d52a3.

wesleyboar commented on user-guide/mkdocs.yml:
```diff
@@ -1,39 +1,34 @@
       - before: '#data-depotcuration-office-hours'
         after: https://www.designsafe-ci.org/facilities/virtual-office-hours/
-    analytics:
-      gtag: G-D96RT1T24K
```
Whoops. Put this back in.

wesleyboar replied:
Restored: a0775c69

wesleyboar reviewed (commented):
Almost done!

wesleyboar committed:
chore: use tacc-docs v0.2.0

wesleyboar committed:
chore: use tacc-docs v0.3.1

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
fix: DES-2469 accent color

Uses Core-Styles v3 pre-alpha-esque code.

wesleyboar committed:
fix: DES-2469 restore lost analytics tag

wesleyboar committed:
docs: DES-2469 when to run tacc script

wesleyboar reviewed (commented):
Resolved issues in dev review \# 2.

wesleyboar committed:
fix: DES-2469 do not use `<hr>` wiht inline styles

wesleyboar committed:
feat: tacc-docs v0.4

https://github.com/TACC/TACC-Docs/releases/tag/v0.4.0

wesleyboar committed:
test: support descriptive title without inline CSS

GOAL: Replace `font-size:1.5em;` with a solution that is not dependent on font-size.
NOTES: Intent of inline style was to create a title that is descriptive that only shows on page (not in nav).
PROBLEM: Solution uses inline styles.

wesleyboar committed:
feat: add n' use variable BASE_URL for TACC files

wesleyboar committed:
Merge branch 'main' into task/tacc-docs-shared-docker

wesleyboar committed:
fix: define container name explicitely

https://github.com/DesignSafe-CI/DS-User-Guide/pull/1#discussion_r1608611302

wesleyboar committed:
feat: support descriptive title without inline CSS

wesleyboar committed:
fix: ADCIRC 404

wesleyboar committed:
fix: move keywords to expected location

we'll do more of this later

wesleyboar committed:
docs: improve syntax in a comment

wesleyboar committed:
fix: img vs svg logo spacing

wesleyboar committed:
fix:  use new logo (svg)

wesleyboar commented on user-guide/docs/usecases/lowes/usecase.md:
```diff
@@ -1,16 +1,18 @@
+/// html | header
+
 ## Seismic Response of Concrete Walls
 
-<span style="font-size:1.5em;">
-**Modeling Reinforced Concrete Walls using Shell Elements in OpenSees and Using Jupyter to Post Process Results**
-</span>
+Modeling Reinforced Concrete Walls using Shell Elements in OpenSees and Using Jupyter to Post Process Results
+
+///
```
**Why?** To make long title and short/nav title look coupled (and short/nav title less important).

<details>

| before | after |
| - | - |
| ![before](https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/f454a703-ec6a-4d9b-a388-7f55cde08ee3) | ![after](https://github.com/DesignSafe-CI/DS-User-Guide/assets/62723358/e4668754-33c2-4d26-a52f-5255c649359b) |

Before me:

1. The navigation had been using short titles, and the pages had been using long titles.
2. The documents were concatenated and navigation changed to match TACC user experience.
3. That caused navigation titles to differ from page titles.
4. A solution was implemented to put long and short titles on the page, so:
    - User could see the short nav title on the page, to check navigation.
    - User could see the long title on the page, to know the full name.

After me:

5. TACC styles made the short title and long title look less coupled.
6. I changed the markup so I could add CSS to:
    - visually couple the tiles
    - make short title be less important

After this PR:

7. We will review reverting to the old navigation (step 1).

</details>

wesleyboar commented:
> [!IMPORTANT]
> All reviews and tasks completed for (initial method of) reliance on TACC-Docs.
